### PR TITLE
chore: Test IOx with proposed changes to DataFusion statistics

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
Make sure IOx continues to run with the proposed change to statistics from https://github.com/apache/arrow-datafusion/pull/965 from @rdettai

# Changes
1. Update DataFusion IOx to use branch in https://github.com/apache/arrow-datafusion/pull/965
2. Update code to use new API

### CODE REVIEW

File: query/src/provider.rs [hunk 1]
```diff
@@ -267,30 +264,6 @@ impl<C: QueryChunk + 'static> TableProvider for ChunkTableProvider<C> {
         Ok(plan)
     }
 
-    fn statistics(&self) -> Statistics {
```
Comment (alamb) [inline query/src/provider.rs:270]: this code is moved into `query/src/provider/physical.rs`

File: query/src/provider/deduplicate.rs [hunk 2]
```diff
@@ -236,6 +237,20 @@ impl ExecutionPlan for DeduplicateExec {
     fn metrics(&self) -> Option<MetricsSet> {
         Some(self.metrics.clone_inner())
     }
+
+    fn statistics(&self) -> Statistics {
+        // TODO: we should acount for overlaps at this point -- if
```
Comment (alamb) [inline query/src/provider/deduplicate.rs:242]: this is is the existing, if a little clunky (aka suspicious) behavior

File: query/src/statistics.rs [hunk 3]
```diff
@@ -39,20 +39,27 @@ pub(crate) fn df_from_iox(schema: &Schema, summary: &TableSummary) -> DFStatisti
         .map(|(i, (_, field))| (field.name(), i))
         .collect::<hashbrown::HashMap<_, _>>();
 
-    let mut columns = summary.columns.iter().collect::<Vec<_>>();
+    let mut columns: Vec<(&ColumnSummary, &usize)> = summary
```
Comment (alamb) [inline query/src/statistics.rs:42]: As projection (e.g a restriction to a subset of columns) is done during physical planning, I needed to update this code to allow for selecting a subset of the columns in the statistics

File: query_tests/cases/in/stats_plans.expected [hunk 4]
```diff
@@ -4,9 +4,9 @@
 | plan_type     | plan                                                        |
 +---------------+-------------------------------------------------------------+
 | logical_plan  | Projection: #COUNT(UInt8(1))                                |
-|               |   Projection: UInt64(3) AS COUNT(Uint8(1))                  |
-|               |     EmptyRelation                                           |
-| physical_plan | ProjectionExec: expr=[COUNT(UInt8(1))@0 as COUNT(Uint8(1))] |
+|               |   Aggregate: groupBy=[[]], aggr=[[COUNT(UInt8(1))]]         |
```
Comment (alamb) [inline query_tests/cases/in/stats_plans.expected:7]: The logical plan no longer shows the optimization as it is now done in the physical plan (which still has the optimization).

Looks like @rdettai has also fixed a bug in the column naming `Uint8` --> `UInt8` :bowtie:

File: query/src/provider/physical.rs [hunk 5]
```diff
@@ -22,9 +22,9 @@ use super::adapter::SchemaAdapterStream;
 #[derive(Debug)]
 pub(crate) struct IOxReadFilterNode<C: QueryChunk + 'static> {
     table_name: Arc<str>,
-    /// The desired output schema (includes selection_
+    /// The desired output schema (includes selection)
     /// note that the chunk may not have all these columns.
-    schema: SchemaRef,
+    iox_schema: Arc<Schema>,
```
Comment (NGA-TRAN) [inline query/src/provider/physical.rs:27]: SchemaRef and Arc<Schea> are the same. And I agree this new code looks clearer with the name `iox_schema`

**Review by NGA-TRAN — APPROVED** [review]
LGTM

File: query/src/provider/physical.rs (hunk 5 above)
Comment (NGA-TRAN) [reply query/src/provider/physical.rs:27]: make sense as you also removed  the `.to_arrow`

**Comment by alamb** [comment]
Obseleted by https://github.com/influxdata/influxdb_iox/pull/2521

**Pull request closed without merging**